pubsub/google/cloud/pubsub_v1/publisher/client.py

```diff
@@ -407,6 +396,13 @@ def _wait_and_commit_sequencers(self):
             self._commit_thread = None
 
     def _commit_sequencers(self):
+        """ Clean up finished sequencers and commit the rest. """
+        finished_sequencer_keys = [
+            k[0] for k in self._sequencers.items() if k[1].is_finished()
+        ]
```

Comment:
It would IMO be a bit more readable if we directly unpacked the items, rather than unpacking them with numeric indexes:
```py
finished_sequencer_keys = (
    key for key, sequencer in self._sequencers.items() if sequencer.is_finished()
)
```

Reply:
Using tuple destructuring to get the key & sequencer is a good idea.

I'd prefer not to allocate another object, so wont make the second change. Thanks for reminding me that dict comprehensions exist; I usually only use list comprehensions.